backend/internal/api/service_handlers.go

```diff
@@ -222,14 +223,56 @@ func (h *ServiceHandler) getViewportNodes(c *gin.Context) {
 
 // parseVault triggers a vault parsing operation
 func (h *ServiceHandler) parseVault(c *gin.Context) {
-	// TODO: Implement when VaultService is available
-	c.JSON(http.StatusNotImplemented, gin.H{"error": "Vault parsing not yet implemented"})
+	// Use request context
+	ctx := c.Request.Context()
+
+	// Check if vault service is available
+	if h.vaultService == nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Vault service not configured"})
+		return
+	}
+
+	// Check if parse is already running
+	status, err := h.vaultService.GetParseStatus(ctx)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get parse status"})
+		return
+	}
+
+	if status.Status == "parsing" {
+		c.JSON(http.StatusConflict, gin.H{"error": "Parse already in progress"})
```

Comment:
TODO(CL): Consider rate limiting for parseVault endpoint. Since parsing is resource-intensive:
1. Limit to 1 parse per minute per client
2. Add queue position in conflict response
3. Consider API key-based limits for different tiers

Example conflict response:
```json
{
  "error": "Parse already in progress",
  "queue_position": 3,
  "estimated_wait_time_seconds": 180
}
```